### PR TITLE
add mvp architecture

### PR DESCRIPTION
**Pull request by DiegoDiazGlobant**
The value of the buttons is displayed on the screen. Not concatenated yet.

Nexus S - preview

https://user-images.githubusercontent.com/75440409/104958641-29c72d00-59af-11eb-9486-d96311f3cd67.mp4

### CODE REVIEW

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.example.calculator.mvp.presenter;
+
+import com.example.calculator.mvp.contract.CalculatorContract;
+import com.example.calculator.mvp.model.CalculatorModel;
+import com.example.calculator.mvp.view.CalculatorView;
+
+public class CalculatorPresenter implements CalculatorContract.Presenter {
+
+    private CalculatorModel model;
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: these variables must be of the contract type

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    private CalculatorView view;
+
+    public CalculatorPresenter(CalculatorModel model, CalculatorView view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: same here in the parameters

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java (hunk 2 above)
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java:12]: you should use de contract types here

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java (hunk 1 above)
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: you should use de contract types here

Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: Yes! solved.

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java (hunk 2 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: Solved!

File: app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java (hunk 1 above)
Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java]: Solved!

**Review by lucianoeli — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 3]
```diff
@@ -1,14 +1,42 @@
...
+        binding.buttonOne.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonOne.getText().toString()));
+        binding.buttonTwo.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonTwo.getText().toString()));
+        binding.buttonThree.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonThree.getText().toString()));
+        binding.buttonFour.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonFour.getText().toString()));
+        binding.buttonFive.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonFive.getText().toString()));
+        binding.buttonSix.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonSix.getText().toString()));
+        binding.buttonSeven.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonSeven.getText().toString()));
+        binding.buttonEight.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonEight.getText().toString()));
+        binding.buttonNine.setOnClickListener(view -> presenter.onButtonPressed(binding.buttonNine.getText().toString()));
```
Comment (nicolbidone) [inline app/src/main/java/com/example/calculator/MainActivity.java]: you can make this more legible implementing 
presenter.onButtonPressed(binding.buttonNine.getText().toString())
on a separate method

Comment (DiegoDiazGlobant) [reply app/src/main/java/com/example/calculator/MainActivity.java]: Thanks! Refactored

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/example/calculator/MainActivity.java [hunk 4]
```diff
@@ -22,21 +23,25 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        setUpClickListenerToButton(binding.buttonThree);
+        setUpClickListenerToButton(binding.buttonFour);
+        setUpClickListenerToButton(binding.buttonFive);
+        setUpClickListenerToButton(binding.buttonSix);
+        setUpClickListenerToButton(binding.buttonSeven);
+        setUpClickListenerToButton(binding.buttonEight);
+        setUpClickListenerToButton(binding.buttonNine);
+    }
+
+    private void setUpClickListenerToButton(Button buttonPressed) {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/calculator/MainActivity.java:44]: NICEE!

**Review by benjaglobant — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Review by LuchoElissondoGlobant — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Pull request merged**